### PR TITLE
Adding unit tests for server & database

### PR DESCRIPTION
**Pull request by gektin**
**/server**
I split up my app from my index file in order to be able to run tests using a different port more easily. This was just moving some code around and using requires. In terms of new code, there's a catch block for the server now to handle server errors, as well as a 404 block for client errors. 

**/spec**
I added a db test to check there are 100 entries and that the 100th entry is the expected one.

I added two server tests to check the routes I'm serving right now. A happy path (bundle 49) and a sad path (bundle 149), and those tests are both passing.

**other files**
Mostly nits like adding new lines and working on logging and similar changes.

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  .then((bundle) => {
+    if (bundle) {
+      res.send(JSON.stringify(bundle));
+    }
+  })
+  .catch((err) => {
+    res.status(500).send('something went wrong on our end; wait a bit & try again');
+  });
+});
+
```
Comment (redperezz) [inline server/app.js:38]: Very nice error handling!! 👌🏽✓

File: server/index.js [hunk 2]
```diff
@@ -1,53 +1,19 @@
...
-        id: 100,
-      },
-    })
-    .then((results) => {
-      console.log('results: ', results.dataValues);
-    });
-  })
   .catch((err) => {
     console.log('your error: ', err);
   });
```
Comment (redperezz) [inline server/index.js:18]: Great separation with your db/index file! I could be blind but I'm not seeing the use of the Bundles const on line 2. If it is ignore this comment lol.

File: spec/client.spec.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
-require('iconv-lite').encodingExists('foo')
+require('mysql2/node_modules/iconv-lite').encodingExists('foo');
 
 describe('basic rendering', () => {
   test('it loads', () => {
     expect(true).toBe(true);
   })
```
Comment (redperezz) [inline spec/client.spec.js:6]: ✓

File: spec/db.spec.js [hunk 4]
```diff
@@ -1,25 +1,41 @@
...
         expect(results.dataValues.value).toBe(183);
         expect(results.dataValues.logoURL).toBe('https://unsplash.com/photos/YcfCXxo7rpc');
         expect(results.dataValues.helper).toBe('@causam');
         expect(results.dataValues.minimum).toBe(6);
-      })
-    })
+      });
+    };
   });
 });
```
Comment (redperezz) [inline spec/db.spec.js:41]: Looks great and I agree with this test case! ✓

File: spec/server.spec.js [hunk 5]
```diff
@@ -1,7 +1,29 @@
...
+  test('it should respond to a GET to \'/bundleInfo/bundleId\' for a bundle that doesn\'t exist', (done) => {
+    return request(app)
+      .get('/bundleInfo/149')
+      .then((response) => {
+        expect(response.statusCode).toBe(404);
+        expect(response.error.text).toBe('no such bundle, try a bundle from 1 - 100');
+      })
+      .then(done());
+  });
+});
```
Comment (redperezz) [inline spec/server.spec.js:29]: Very nice test cases for your error handling in server/app.js! Great work! ✓

**Review by redperezz — APPROVED** [review]
Great work! Didn't spot any syntax errors or logical concerns 👍🏽

File: server/index.js (hunk 2 above)
Comment (gektin) [reply server/index.js:18]: Deleted.

**Pull request merged**